pennylane/interfaces/tf.py

```diff
@@ -110,17 +113,17 @@ def grad(grad_output, **tfkwargs):
             grad_input = unflatten(temp.flat, args)
 
             if isinstance(grad_input, list):
-                grad_input = [tf.convert_to_tensor(i) for i in grad_input]
+                grad_input = [tf.convert_to_tensor(i, dtype=dtype) for i in grad_input]
             elif isinstance(grad_input, tuple):
-                grad_input = tuple(tf.convert_to_tensor(i) for i in grad_input)
+                grad_input = tuple(tf.convert_to_tensor(i, dtype=dtype) for i in grad_input)
             else:
-                grad_input = tf.convert_to_tensor(grad_input)
+                grad_input = tf.convert_to_tensor(grad_input, dtype=dtype)
 
             if variables is not None:
                 return grad_input, variables
 
             return grad_input
 
-        return res, grad
+        return tf.convert_to_tensor(res, dtype=dtype), grad
```

Comment:
@josh146, one thing that wasn't clear to me is where `tf.numpy_function` would be used. Were you thinking something like
```python
return tf.numpy_function(res), tf.numpy_function(grad)
```

Reply:
It would be used up on line 103; instead of 

```python
jacobian = qnode.jacobian(args, kwargs)
```

we instead would have

```python
jacbobian = tf.numpy_function(lambda inpt: qnode.jacobian(inpt, kwargs), [args], dtype) 
```

Reply:
I think this might be non-trivial to change though!